src/prommis/nanofiltration/costing/diafiltration.py

```diff
@@ -0,0 +1,713 @@
+#########################################################
+# MSContactor flowsheet code adapted from Andrew Lee.
+#########################################################
+
+
+import pyomo.environ as pyo
+from pyomo.network import Arc
+
+from idaes.core.util.model_diagnostics import DiagnosticsToolbox
+
+from idaes.core import (
+    Component,
+    declare_process_block_class,
+    FlowsheetBlock,
+    MaterialBalanceType,
+    MaterialFlowBasis,
+    Phase,
+    PhysicalParameterBlock,
+    StateBlock,
+    StateBlockData,
+    UnitModelBlock,
+    UnitModelCostingBlock,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, propagate_state
+from prommis.nanofiltration.costing.diafiltration_cost_model import (
+    DiafiltrationCosting,
+    DiafiltrationCostingData,
+)
+
+from idaes.models.unit_models import (
+    MSContactor,
+    MSContactorInitializer,
+    Mixer,
+    MixingType,
+    MomentumMixingType,
+    MixerInitializer,
+)
+
+
+class _StateBlock(StateBlock):
```

Comment:
To be a little more general, it might be good to move the property package classes to a separate module. That way, they can be reused in other places if/when necessary, as well as keeping modules smaller.